lib/manageiq/providers/hawkular/resource_collection.rb

```diff
@@ -0,0 +1,40 @@
+require 'generic_view_mapper'
+require_relative 'entity_mapper'
+require_relative 'entities/middleware_server'
+require_relative 'entities/wildfly_server'
+
+module ManageIQ
+  module Providers
+    module Hawkular
+      class ResourceCollection
+        attr_reader :data
+
+        def initialize(resources)
+          @data = resources
+        end
+
+        def entities
+          @entities ||= data.flat_map do |d|
+            data = EntityMapper.map(d)
+            GenericViewMapper.matcher.find_entity_for(data).new(data)
+          end
+        end
+
+        def prepared
+          servers = entities.select { |x| x.kind_of?(Entities::MiddlewareServer) }
+          resources = (entities - servers).group_by(&:feed)
+
+          servers.each do |server|
+            server.relationships = resources[server.feed]
+
+            resources[server.feed]
+              .inject({}) { |accum, el| accum.merge(el.properties) }
+              .merge!(server.properties)
```

Comment:
What's this doing? I don't see where the accumulated properties hash ends, so I think it just gets collected by the garbage collector.

Reply:
This is done for view-related purposes, to cover cases such as when we need to show if a server is immutable or not, since it’s not being defined on the Server entity, but on the OperatingSystem resources.